projects/wvr-elements/src/lib/wvr-modal/wvr-modal.component.ts

```diff
@@ -149,32 +149,22 @@ export class WvrModalComponent extends WvrBaseComponent implements OnInit {
     this.store.dispatch(ModalActions.openModal({ id: this.modalId }));
   }
 
-  /** This provides background, border and text color properties based on the them variant provided . */
-  additionalClasses(value): string {
-    let additionalClasses = '';
-    switch (value) {
-      case 'header':
-        additionalClasses = this.modalHeaderThemeVariant ?
-          `bg-${this.modalHeaderThemeVariant} border-${this.modalHeaderThemeVariant} ${this.getTextColor(this.modalHeaderThemeVariant)}` :
-          this.themeVariant ?
-            `bg-${this.themeVariant} border-${this.themeVariant} ${this.getTextColor(this.themeVariant)}` :
-            'bg-light text-dark';
-        break;
-      case 'footer':
-        additionalClasses = this.modalFooterThemeVariant ?
-          `bg-${this.modalFooterThemeVariant} border-${this.modalFooterThemeVariant} ${this.getTextColor(this.modalFooterThemeVariant)}` :
-          this.themeVariant ?
-            `bg-${this.themeVariant} border-${this.themeVariant} ${this.getTextColor(this.themeVariant)}` :
-            'bg-light text-dark';
-        break;
-      default:
-    }
-
-    return additionalClasses;
+  get additionalHeaderClasses(): string {
```

Comment:
I wonder if this could be further improved.
Would it be reasonable to create a function that accepts an argument to perform this task for both header and footer functions?

Such as: `additionalClasses(themeVariant?: string)`.

Then this could be coded once in the function and have `modalHeaderThemeVariant` and `modalFooterThemeVariant` passed to as appropriate?

Reply:
It must be a getter without arguments for not to violate lint error `no call expression errors` in templates.

Reply:
template-no-call-expression

http://codelyzer.com/rules/no-template-call-expression/